### PR TITLE
fix: judge rm targets with a platform-neutral join

### PR DESCRIPTION
**Pull request by vitali87**
## What

Three fixes to the `rm` guard in `codebase_rag/tools/shell_command.py`. The first is the one this PR opened with; the other two are real escapes that local review found while verifying it, both of which delete files outside the project on this host's BSD `rm`.

### 1. Platform-neutral absoluteness test

`_is_dangerous_rm_path` tested absoluteness with `path_arg.startswith("/")`, the POSIX-only idiom that #1662 removed from the git repo-location guard:

```diff
-            if path_arg.startswith("/"):
-                resolved = Path(path_arg).resolve()
-            else:
-                resolved = (project_root / path_arg).resolve()
+            resolved = (project_root / path_arg).resolve()
```

`Path.__truediv__` applies the platform's own rule: an absolute target replaces the root, a relative one lands under it.

### 2. Operands were collected by leading dash, not by rm's own rule

The guard filtered out every dash-prefixed token as an option, so these two spellings skipped the containment check entirely and deleted the victim:

```
rm -r -- -x/../../outside/victim     # -- ends option parsing
rm a  -x/../../outside/victim        # options end at the first operand
```

`_rm_operands` now follows POSIX: option parsing ends at the first operand or at `--`, whichever comes first, and every token from there on is a path even when it starts with a dash. BSD `rm` (macOS) implements that literally. GNU `rm` instead permutes and rejects `-x/` as a bad option, so taking the POSIX reading classifies *more* tokens as paths than GNU would — the containment check only ever errs towards refusing.

### 3. Operand text was being read as flags

`_is_dangerous_rm` scanned every dash-prefixed token for `r` and `f`, so `rm -r -- -in-root-file` was blocked as if it were `rm -rf` — the filename happens to contain both letters. It now shares the same option/operand split via `_rm_options`, derived from `_rm_operands` so the two helpers cannot drift apart.

## Impact

- **The `startswith("/")` change is production-neutral.** `ShellCommander` resolves the root in its constructor, so relative targets already landed under a drive-qualified root on Windows. POSIX verdicts are provably identical (`/a / "/b" == /b`); checked old vs new on 24 spellings with zero differences.
- **The two operand fixes are real.** Both spellings above delete `outside/victim` on `main` and are refused at the `execute()` surface here, in normal and yolo mode.
- **Nothing that `main` blocked now gets through.** The 714 argv combinations whose verdict changed are all the same class: dash tokens after the first operand or after `--`, which real `rm` treats as paths too. Confirmed against `/bin/rm`: `rm a -rf` in a directory containing an `-rf` file deletes that file and leaves directory `a` intact.
- **Windows.** A rooted, drive-less target such as `/x` now resolves on the root's drive (the drive `rm` runs on) rather than the process's. Every such case is still blocked.

## Verification

- `codebase_rag/tests/test_shell_command.py`: **1150 passed, 1 skipped** (macOS, via the host suite lock).
- Differential probe against this host's real `/bin/rm` across 118 argv spellings plus symlink, absolute-path and separator edge cases: **zero escapes**, both legitimate in-root spellings keep their pass.
- Five new tests; three of them fail against `main`'s implementation and pass here.
- Cognitive complexity of `_is_dangerous_rm_path` goes 17 → 15 (S3776 limit is 15).
- `ruff check` / `ruff format --check`: clean.

Independent of #1662; based on `main`.

## Summary by CodeRabbit

- **Bug Fixes**
  - Improved safety checks for potentially dangerous `rm` commands, including options appearing after operands, `--`, and dash-prefixed filenames.
  - Ensured all command operands are checked for project containment, preventing traversal paths from bypassing validation.
  - Improved handling of absolute and relative paths, including rooted, drive-less paths on Windows.
  - Fixed validation of Git targets located on different drives.

- **Tests**
  - Added coverage for GNU-style option handling and valid paths within the project.

### CODE REVIEW

**Comment by vitali87** [comment]
@greptileai review

**Comment by vitali87** [comment]
@coderabbitai review

File: codebase_rag/tools/shell_command.py [hunk 1]
```diff
@@ -146,11 +146,13 @@ def _is_dangerous_rm_path(cmd_parts: list[str], project_root: Path) -> tuple[boo
...
+        # absolute target replaces the root outright, a relative one lands
+        # under it. A `startswith("/")` check is POSIX-only: on Windows a
+        # rooted, drive-less target such as `/x` would resolve on the Python
+        # process's drive rather than the root's, the drive rm runs on.
         try:
-            if path_arg.startswith("/"):
-                resolved = Path(path_arg).resolve()
-            else:
-                resolved = (project_root / path_arg).resolve()
+            resolved = (project_root / path_arg).resolve()
```
[the remark this answers is not in the document]
Comment (vitali87) [reply codebase_rag/tools/shell_command.py:197]: Confirmed, and pre-existing: on `main` the old `Path(path_arg).resolve()` branch produces the same `resolved == project_root` result for an absolute root spelling, and `subdir/..` took the join branch already, so this PR neither introduces nor changes it. It is filed as #1667 with the suggested fix (reject `resolved == project_root` before the containment check, tests for the absolute spelling and `subdir/..`). This PR stays scoped to the `startswith("/")` idiom, the same change #1662 makes to the git guard, so the two diffs are reviewable as one idea.

**Comment by vitali87** [comment]
@greptileai review

**Comment by vitali87** [comment]
@coderabbitai review

**Comment by vitali87** [comment]
@greptileai review

**Comment by vitali87** [comment]
@coderabbitai review

**Comment by vitali87** [comment]
@greptileai review

**Comment by vitali87** [comment]
@coderabbitai review
